nservicebus/upgrades/throughput-tool.md

```diff
@@ -0,0 +1,143 @@
+1. Execute `az set --subscription {SubscriptionId}`, where `SubscriptionId` is a Guid matching the subscription id that contains the Azure Service Bus namespace.
+1. In the Azure Portal, go to the Azure Service Bus namespace, click **Properties** in the side navigtation, and then copy the `Id` value, which will be needed to run the tool. The `Id` value should have a format similar to `/subscriptions/{Guid}/resourceGroups/{rsrcGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}`.
+
+Once these prerequisites are comlpete, execute the tool with the resource id of the Azure Service Bus namespace, as in this example:
+
+```shell
+particular-throughput-tool azureservicebus --resourceId /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/my-resource-group/providers/Microsoft.ServiceBus/namespaces/my-asb-namespace --outputPath throughput-report.json
+```
+
+The tool may open additional terminal windows, which are Powershell processes gathering the data from the Azure CLI.
+
+Unlike other query options, using Azure Service Bus metrics allows the tool to capture the last 30 days worth of data, which means that results are returned instantly.
+
+#### Options
+
+All options are required:
+
+| Option | Description |
+|-|-|
+| <nobr>`--resourceId`</nobr> | The resource ID of the Azure Service Bus namespace, which can be found in the Azure Portal as described above. |
+| <nobr>`--outputPath`</nobr> | A local path where the JSON report file should be generated. |
+
+### RabbitMQ
+
+Once installed, execute the tool with the RabbitMQ management URL, as in this example:
+
+```shell
+particular-throughput-tool rabbitmq --apiUrl http://localhost:15672 --outputPath throughput-report.json
+```
+
+The tool will prompt for the username and password to access the RabbitMQ management interface. After that, it will take its initial reading, then sleep for 24 hours before taking its final reading and generating a report.
+
+#### Options
+
+All options are required:
+
+| Option | Description |
+|-|-|
+| <nobr>`--apiUrl`</nobr> | The URL for the RabbitMQ management site. |
+| <nobr>`--outputPath`</nobr> | A local path where the JSON report file should be generated. |
+
+### SQL Transport
+
+Once installed, execute the tool with the database connection string used by SQL Server endpoints, as in this example:
+
+```shell
+particular-throughput-tool sqlserver --connectionString "Server=SERVER;Database=DATABASE;User=USERNAME;Password=PASSWORD;" --outputPath throughput-report.json
+```
+
+The tool will run for slightly longer than 24 hours in order to capture a beginning and ending `RowVersion` value for each queue table. A value can only be detected when a message is waiting in the queue to be processed, and not from an empty queue, so multiple queries may be required. The tool will use a backoff mechanism to avoid putting undue pressure on the SQL Server instance.
```

Comment:
As above, will data collected be written into temporary files to mitigate the risk of a crash?